### PR TITLE
Wrap dart:convert to track utf8 decode failures

### PR DESCRIPTION
**Pull request by jonahwilliams**
Creates a tools specific implementation of Utf8Decoder that allows malformed utf8 and then checks for \u{FFFD}. If that is encountered, an error is thrown with more context.

Adds an unauthorized imports test to cover the flutter_tools lib directory.

https://github.com/flutter/flutter/issues/15646

### CODE REVIEW

**Comment by jonahwilliams** [comment]
cc @Hixie

File: packages/flutter_tools/lib/src/android/android_device.dart [hunk 1]
```diff
@@ -681,7 +681,8 @@ class _AdbLogReader extends DeviceLogReader {
       _timeOrigin = null;
     runCommand(device.adbCommandForDevice(args)).then<void>((Process process) {
       _process = process;
-      const Utf8Decoder decoder = Utf8Decoder(allowMalformed: true);
+      // Previously this line specified allowMalformed: true.
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/android/android_device.dart]: Avoid "history of the code" comments. Maybe explain why reportErrors is true, instead.

File: packages/flutter_tools/lib/src/android/android_workflow.dart [hunk 2]
```diff
@@ -257,12 +257,14 @@ class AndroidLicenseValidator extends DoctorValidator {
     );
     process.stdin.write('n\n');
     final Future<void> output = process.stdout
-      .transform<String>(const Utf8Decoder(allowMalformed: true))
+      // Previously this line specified allowMalformed: true.
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/android/android_workflow.dart]: ditto here and below.

File: packages/flutter_tools/lib/src/convert.dart [hunk 3]
```diff
@@ -0,0 +1,47 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/convert.dart]: 2019

File: packages/flutter_tools/lib/src/convert.dart [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert' hide utf8;
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/convert.dart]: Neat! Maybe add a comment here explaining what this does.

**Review by zanderso — APPROVED** [review]

**Pull request merged**